cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java

```diff
@@ -91,21 +95,21 @@ public void execute(Project project) {
         } catch (IOException e) {
             throw createLauncherException(e.getMessage());
         }
-
+        List<Diagnostic> jarResolverDiagnostics = null;
         try {
             PackageCompilation pkgCompilation = project.currentPackage().getCompilation();
             JBallerinaBackend jBallerinaBackend = JBallerinaBackend.from(pkgCompilation, JvmTarget.JAVA_11);
             long start = 0;
             if (project.buildOptions().dumpBuildTime()) {
                 start = System.currentTimeMillis();
             }
-
             if (project.buildOptions().nativeImage() && project.buildOptions().cloud().equals("")) {
                 jBallerinaBackend.emit(JBallerinaBackend.OutputType.GRAAL_EXEC, executablePath);
             } else {
                 jBallerinaBackend.emit(JBallerinaBackend.OutputType.EXEC, executablePath);
             }
 
+            jarResolverDiagnostics = new ArrayList<>(jBallerinaBackend.jarResolver().diagnosticResult().diagnostics());
```

Comment:
Can't we assign `jBallerinaBackend.jarResolver().diagnosticResult().diagnostics()` directly without creating a new list

Reply:
> jBallerinaBackend.jarResolver().diagnosticResult().diagnostics()

This is a collection, not a list. That's why we need to create a list.

Reply:
> We can print the diagnostics here. This will eliminate the need for null check at L142

If we do that it gets inserted in between, `Generating executable` and `target path`. That's why it is moved after those lines.

Reply:
We can remove the null check since its redundant.